From c0778d1eb7829ad92288004ae5bc18206aa7af43 Mon Sep 17 00:00:00 2001
From: chenzhen <chenzhen@rock-chips.com>
Date: Wed, 30 Nov 2016 08:48:43 +0800
Subject: [PATCH] MALI: utgard: RK: use late_initcall_sync instead of
 module_init

Some dependences of mali device driver should be initialized first.

Change-Id: I76f1d8b029345801bf0a68266889ec1c5a28b524
Signed-off-by: chenzhen <chenzhen@rock-chips.com>
---
 drivers/gpu/arm/mali400/mali/linux/mali_kernel_linux.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/arm/mali400/mali/linux/mali_kernel_linux.c b/drivers/gpu/arm/mali400/mali/linux/mali_kernel_linux.c
index 29964d199163..ae4fad8d7588 100755
--- a/drivers/gpu/arm/mali400/mali/linux/mali_kernel_linux.c
+++ b/drivers/gpu/arm/mali400/mali/linux/mali_kernel_linux.c
@@ -970,8 +970,7 @@ static int mali_ioctl(struct inode *inode, struct file *filp, unsigned int cmd,
 	return err;
 }
 
-
-module_init(mali_module_init);
+late_initcall_sync(mali_module_init);
 module_exit(mali_module_exit);
 
 MODULE_LICENSE(MALI_KERNEL_LINUX_LICENSE);
-- 
2.35.3

